src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs

```diff
@@ -1950,6 +1961,23 @@ public bool Equals(ReadOnlySpan<char> span)
         /// are removed from this builder.
         /// </remarks>
         public StringBuilder Replace(string oldValue, string? newValue, int startIndex, int count)
+        {
+            ArgumentException.ThrowIfNullOrEmpty(oldValue);
```

Comment:
This can just be ThrowIfNull. The called method has the check for empty.

Reply:
It would change the exception message (I think) in case it's an empty string (`SR.Arg_EmptySpan` instead of `SR.Argument_EmptyString`), I don't know if that would be an issue? It might be confusing to the user who's using strings to get an exception message indicating an empty span?